x-pack/plugins/global_search_providers/public/providers/get_app_results.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import levenshtein from 'js-levenshtein';
+import { PublicAppInfo, PublicLegacyAppInfo } from 'src/core/public';
+import { GlobalSearchProviderResult } from '../../../global_search/public';
+
+export const getAppResults = (
+  term: string,
+  apps: Array<PublicAppInfo | PublicLegacyAppInfo>
+): GlobalSearchProviderResult[] => {
+  return (
+    apps
+      // remove disabled and chromeless app (to avoid returning the login page for example)
+      .filter((app) => app.status === 0 && (app.legacy === true || app.chromeless !== true))
+      .map((app) => ({ app, score: scoreApp(term, app) }))
+      .filter(({ score }) => score > 0)
+      .map(({ app, score }) => appToResult(app, score))
+  );
+};
+
+export const scoreApp = (term: string, { title }: PublicAppInfo | PublicLegacyAppInfo): number => {
+  term = term.toLowerCase();
+  title = title.toLowerCase();
+
+  // shortcuts to avoid calculating the distance when there is an exact match somewhere.
+  if (title === term) {
+    return 100;
+  }
+  if (title.startsWith(term)) {
+    return 90;
+  }
+  if (title.includes(term)) {
+    return 75;
+  }
+  const length = Math.max(term.length, title.length);
+  const distance = levenshtein(term, title);
+
+  // maximum lev distance is length, we compute the match ratio (lower distance is better)
+  const ratio = Math.floor((1 - distance / length) * 100);
+  if (ratio >= 60) {
+    return ratio;
+  }
+  return 0;
```

Comment:
Note that all the scoring logic in `scoreApp` is totally arbitrary, as we have to implement it ourselves. 

I'm still unsure about the need to reply on levenshtein distance when there is not an exact match. There are not a lot of apps, so the computation time should be negligible though.

@elastic/kibana-core-ui WDYT? Are exact (sub-)matches sufficient when searching for applications, or do you think the levenshtein scoring makes sense here?

Reply:
It doesn't seem to complicate much, so I'd rather not use exact match...

Thinking about non-English/non-native English speakers trying to use Kibana when we don't translate our app names, this seems like a nice way to catch typos. 

Some examples:
- Visualize vs Visualise 
- Machine Learning vs Mashine vs Machien (actually, thinking about it, there are like a million ways to spell machine so it still sounds right)
- Swapping any hard "c" for a "k" (e.g., Dis**c**over, **C**anvas, Metri**c**s)

Reply:
Maybe not an immediate thing, but maybe worth opening a ticket for, it could be nice to translate the app names for search (e.g., someone translates the word "Discover" into their language and searches that, maybe we could still surface Discover then).

Reply:
I haven't dug in deep, but I think our existing setup should work with translated app title strings as well. Those are translated before they enter the Core system so we should always be comparing search terms to the translated strings. Is that right @pgayvallet ?

Reply:
>  Those are translated before they enter the Core system so we should always be comparing search terms to the translated strings

That is correct. As long as the title of the app is a translated string, we will be performing the comparison on the translated title (and would not really be able to do otherwise without adding a new field to `AppBase`).